opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -110,87 +117,104 @@ message TimeSeries {
 message HistogramValue {
-  // The number of values in the population. Must be non-negative. This value
-  // must equal the sum of the values in bucket_counts if a histogram is
+  // count is the number of values in the population. Must be non-negative. This value
+  // must be equal to the sum of the "count" fields in buckets if a histogram is
   // provided.
   int64 count = 1;
 
-  // The sum of the values in the population. If count is zero then this field
-  // must be zero.
+  // sum of the values in the population. If count is zero then this field
+  // must be zero. This value must be equal to the sum of the "sum" fields in buckets if
+  // a histogram is provided.
   double sum = 2;
 
-  // A Histogram may optionally contain the distribution of the values in the
-  // population. The bucket boundaries are described by BucketOptions.
-  message BucketOptions {
-    oneof type {
-      // Bucket with explicit bounds.
-      Explicit explicit = 1;
-    }
-
-    // Specifies a set of buckets with arbitrary upper-bounds.
-    // This defines size(bounds) + 1 (= N) buckets. The boundaries for bucket
-    // index i are:
-    //
-    // [0, bucket_bounds[i]) for i == 0
-    // [bucket_bounds[i-1], bucket_bounds[i]) for 0 < i < N-1
-    // [bucket_bounds[i], +infinity) for i == N-1
-    message Explicit {
-      // The values must be strictly increasing and > 0.
-      repeated double bounds = 1;
-    }
-
-    // TODO: If OpenMetrics decides to support (a, b] intervals we should add
-    // support for these by defining a boolean value here which decides what
-    // type of intervals to use.
-  }
+  // A histogram may optionally contain the distribution of the values in the population.
+  // If that is the case then "bucket_bounds" and "buckets" fields below both must
+  // be defined. Otherwise both fields must be omitted in which case the distribution of
+  // values in the histogram is unknown and only the total count and sum are known.
 
+  // bucket_bounds is an optional field. If present specifies buckets with explicitly
+  // defined bounds. The bucket boundaries are described by this field.
+  //
+  // This defines size(bucket_bounds) + 1 (= N) buckets. The boundaries for bucket
+  // at index i are:
+  //
+  // [0, bucket_bounds[i]) for i == 0
+  // [bucket_bounds[i-1], bucket_bounds[i]) for 0 < i < N-1
+  // [bucket_bounds[i], +infinity) for i == N-1
+  // The values in bucket_bounds array must be strictly increasing and > 0.
   // Don't change bucket boundaries within a TimeSeries if your backend doesn't
   // support this.
-  BucketOptions bucket_options = 3;
+  // If bucket_bounds field is unspecified then bucket bounds are not explicitly defined.
+  // TODO: If OpenMetrics decides to support (a, b] intervals we should add
+  // support for these by defining a boolean value which decides what type of
+  // intervals to use.
+  repeated double bucket_bounds = 3;
```

Comment:
We may want to support the other types of the BucketOptions like Linear. So I don't think this optimization can be done this way.